src/lib.rs

```diff
@@ -458,6 +462,43 @@ trait Signer {
     fn key(&self) -> &Key;
 }
 
+impl<'a> JoseNewAccount<'a> {
+    fn new(account: &NewAccount<'a>, key: &Key, server_url: &str) -> Result<Self, Error> {
+        Ok(Self {
+            contact: account.contact,
+            terms_of_service_agreed: account.terms_of_service_agreed,
+            only_return_existing: account.only_return_existing,
+            external_account_binding: if let Some(eab) = &account.external_account_binding {
+                Some(eab.sign(key, server_url)?)
+            } else {
+                None
+            },
+        })
+    }
+}
+
+impl<'a> ExternalAccountBinding<'a> {
```

Comment:
Style nit: this should go right underneath the defition of `ExternalAccountBinding`.